### PR TITLE
Add tests for docker

### PR DESCRIPTION
**Pull request by Nyarum**
Work for #86 

Before that you need to set a few env's:
- TEST_DOCKER_EMAIL
- TEST_DOCKER_USERNAME
- TEST_DOCKER_PASSWORD

Run tests via Makefile (For example):
- `make test_docker`

Output, like:
```
go test -v ./cmd/docker"
=== RUN   TestDockerLogin
----->  Configuring client
        Project 'Test' with id='57be1db75b0f340007436c8b'
--- PASS: TestDockerLogin (2.14s)
PASS
ok      github.com/iron-io/ironcli/cmd/docker   2.189s
```

### CODE REVIEW

**Comment by Nyarum** [comment]
@nikhilm still in progress.

**Comment by Nyarum** [comment]
@nikhilm R4R.

File: tests/test.js [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (nikhilm) [inline tests/test.js]: Extra file.

File: cmd/docker/docker_test.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+	dockerLogin := NewDockerLogin(settings)
+	dockerLogin.Email = os.Getenv("TEST_DOCKER_EMAIL")
+	dockerLogin.Username = os.Getenv("TEST_DOCKER_USERNAME")
+	dockerLogin.Password = os.Getenv("TEST_DOCKER_PASSWORD")
+
+	err := dockerLogin.Action(settings)
+	if err != nil {
+		t.Error(err)
+	}
```
Comment (nikhilm) [inline cmd/docker/docker_test.go]: Please add tests for failure modes where username is empty and so on.

**Comment by nikhilm** [comment]
@Nyarum Is there a recommended order in which I should review these test PRs?

File: cmd/docker/docker_test.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+	"github.com/iron-io/ironcli/common"
+)
+
+func TestDockerLogin(t *testing.T) {
+	var (
+		settings = &common.Settings{Product: "iron_worker"}
+	)
+
+	common.SetSettings(settings)
```
Comment (nikhilm) [inline cmd/docker/docker_test.go]: Don't like how we call this thing that causes global changes like printing and stuff. Can we just call `common.settingValues()`. Also, settingValues implementation is really weird, can't we have an immutable merge that returns a new settings?

File: cmd/docker/docker_login.go [hunk 4]
```diff
@@ -55,65 +55,53 @@ func NewDockerLogin(settings *common.Settings) *DockerLogin {
...
 	req.Header.Set("Accept", "application/json")
-	req.Header.Set("Authorization", "Basic "+r.TestAuth)
+	req.Header.Set("Authorization", "Basic "+d.TestAuth)
 	req.Header.Set("Content-Type", "application/json")
 
 	res, err := http.DefaultClient.Do(req)
-
 	if err != nil || res.StatusCode != 200 {
-		return errors.New("Docker repo auth failed")
+		return errors.New("Docker repo auth failed, err - " + err.Error())
```
Comment (nikhilm) [inline cmd/docker/docker_login.go]: Convention is `"Docker repo auth failed: %v", err`. Please use `fmt.Errorf`.

File: cmd/docker/docker_login.go [hunk 5]
```diff
@@ -144,3 +132,23 @@ func (r *DockerLogin) Execute(settings *config.Settings, args *map[string]string
...
+	err := d.login()
+	if err != nil {
+		return err
+	}
+
+	auth := map[string]string{
+		"auth": d.RemoteAuth,
+	}
+
+	msg, err := d.Execute(&settings.Worker, &auth)
```
Comment (nikhilm) [inline cmd/docker/docker_login.go]: I don't understand. What is the difference between `login()` and `Execute()`? Why do they do two different HTTP requests and why do we need to do both?

File: cmd/docker/docker_test.go (hunk 3 above)
Comment (Nyarum) [inline cmd/docker/docker_test.go]: @nikhilm all commands do only in one thread without any goroutines. But it'll be cool to improve.

File: cmd/docker/docker_login.go (hunk 5 above)
Comment (Nyarum) [inline cmd/docker/docker_login.go]: @nikhilm this part from the old code. But it not related to tests.

**Comment by Nyarum** [comment]
@nikhilm fixes is added.
